src/atdf/chip.rs

```diff
@@ -31,8 +33,35 @@ pub fn parse(el: &xmltree::Element) -> crate::Result<chip::Chip> {
         })
         .filter(|e| e.name == "interrupt")
         .map(atdf::interrupt::parse)
-        .map(|r| r.map(|int| (int.name.clone(), int)))
-        .collect::<Result<_, _>>()?;
+        .collect::<Result<Vec<_>, _>>()?;
+
+    // Check for duplicate index, merge names if duplicate index exists
+    let mut interrupts = BTreeMap::<usize, chip::Interrupt>::new();
+    for int in interrupts_vec {
+        if let Some(existing_int) = interrupts.get_mut(&int.index) {
+            let old_name = existing_int.name.clone();
+            if let Some(split_idx) = int.name.find('_') {
+                existing_int.name.push_str(int.name.split_at(split_idx).1);
+            } else {
+                existing_int.name.push('_');
+                existing_int.name.push_str(&int.name);
+            }
```

Comment:
I can't say that I am a big fan of this ad-hoc string manipulation because it's very fragile ... But I cannot think of a different solution that does not require bigger changes, so I guess it is fine.  We won't see this 'duplicate interrupt' situation too often anyway, I hope.

Reply:
Same for me. The other option would have been adding a "module-instance" field to the interrupt struct, but than serialization would not work (or needs to exclude the module-instance). This would give a much simpler string manipulation.

Reply:
I saw now that you do the serialization manually,  so actually then an additional field in the interrupt struct would not change anything?

Then we could add a field `module-instance` as `Option<std::string::String>`, and in chip.rs we could append name with module-instance if it exists, and append also the other module-instance in case of duplicate index number.

Reply:
this change would then combine #19 with #20.

Reply:
Let's keep it like it is for now and only fix it if it breaks for someone ;)  No point wasting too much effort on this rare issue ...